### PR TITLE
Add tests and coverage

### PR DESCRIPTION
**Pull request by sttk**
I've added tests and coverage.
With that, the issue #34 will be fixed.

NOTICE: I disabled osx test on travis ci, because its osx server has a lot of backlogs and waits too long time. 
https://www.traviscistatus.com/

### CODE REVIEW

**Comment by sttk** [comment]
Sorry. I mixed unnecessary codes which are for other pr.
I've removed them.

**Comment by phated** [comment]
@sttk it looks like travis fixed the OSX build servers. Can you restore those builds?

File: package.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
     "node": ">= 0.8"
   },
   "scripts": {
-    "test": "jshint lib index.js && jscs lib index.js && mocha -t 5000 -b -R spec test/index"
+    "test": "jshint lib index.js && jscs lib index.js && mocha -t 5000 -b -R spec test/index",
+    "cover": "nyc --reporter=lcov --reporter=text-summary npm test"
```
Comment (phated) [inline package.json]: Can you get this to print the coverage in the CI?  I'm lazy and don't like fetching changesets to run locally.

**Comment by sttk** [comment]
@phated I've modified to enable test on osx and run coverage tool in travis-ci.

**Comment by phated** [comment]
@sttk This is great.  Thank you!

**Comment by sttk** [comment]
@phated Thank you for merging!

**Pull request merged**